### PR TITLE
[Redesign] Add "include prerelease" checkbox and polish search results

### PR DESCRIPTION
**Pull request by joelverhagen**
Fix https://github.com/NuGet/NuGetGallery/issues/4491
Fix https://github.com/NuGet/NuGetGallery/issues/4492

screencapture-localhost-packages-1501777306383 1

### CODE REVIEW

File: src/NuGetGallery/OData/SearchService/SearchAdaptor.cs [hunk 1]
```diff
@@ -25,14 +25,14 @@ public static class SearchAdaptor
         /// </summary>
         internal const int MaxPageSize = 100;
 
-        public static SearchFilter GetSearchFilter(string q, int page, string sortOrder, string context, string semVerLevel)
+        public static SearchFilter GetSearchFilter(string q, int page, string sortOrder, string context, string semVerLevel, bool includePrerelease)
```
Comment (scottbommarito) [inline src/NuGetGallery/OData/SearchService/SearchAdaptor.cs]: nit: In most methods, `includePrerelease` comes before `semVerLevel` in the parameters.

File: src/NuGetGallery/Views/Shared/ListPackages.cshtml [hunk 2]
```diff
@@ -51,4 +60,31 @@
...
     </div>
-</section>
\ No newline at end of file
+</section>
+
+@section bottomScripts {
+    <script type="text/javascript">
+        $(function () {
+            $("#include-prerelease").on('change', function () {
+                var parameters = {};
```
Comment (scottbommarito) [inline src/NuGetGallery/Views/Shared/ListPackages.cshtml]: Is there any way we could simplify this script?

It seems a lot for something that should be innately handled by the form.

Comment (joelverhagen) [inline src/NuGetGallery/Views/Shared/ListPackages.cshtml]: The idea is to create a minimal URL with the knowledge of the server side defaults. This cannot be achieved with a form.

File: src/NuGetGallery/OData/SearchService/SearchAdaptor.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGetGallery/OData/SearchService/SearchAdaptor.cs]: Sure.

**Pull request merged**